makedebian.pl

```diff
@@ -308,10 +315,16 @@
 rm -f /var/lock/subsys/$baseproduct
 
 if [ "$inetd" != "1" ]; then
-	/etc/$baseproduct/stop >/dev/null 2>&1 </dev/null
-	/etc/$baseproduct/start >/dev/null 2>&1 </dev/null
-	if [ "\$?" != "0" ]; then
-		echo "W: $ucproduct server cannot be restarted. It is advised to restart it manually by\nrunning \\"/etc/webmin/restart-by-force-kill\\" when upgrade process is finished"
+	if [ "\$justinstalled" = "1" ]; then
+		/etc/$baseproduct/start >/dev/null 2>&1 </dev/null
+		if [ "\$?" != "0" ]; then
+			echo "E: Webmin server cannot be started. It is advised to start it manually\n   by running \\"webmin force-restart\\" command"
+		fi
+	else
+		/etc/$baseproduct/.post-install >/dev/null 2>&1 </dev/null
```

Comment:
N: In case of upgrade or reinstall always run conditional `.post-install` script, which will simply either run your older code from `/etc/webmin/.start-init` script or in case of a `systemd` systems it will send two signals _SIGCONT_ followed by _SIGHUP_ (making Webmin completely reload `miniserv` configuration).